graphql/src/execution/execution.rs

```diff
@@ -220,6 +205,97 @@ where
                 .and_then(|complexity| total_complexity.checked_add(complexity).ok_or(Overflow))
             })
     }
+
+    // Checks for invalid selections.
+    pub(crate) fn validate_fields(
```

Comment:
It would be good to have a test or two for this so it doesn't break inadvertently if this code ever gets changed.

Reply:
I've added another test.